pkg/internal/kube/store.go

```diff
@@ -432,6 +439,29 @@ func (s *Store) serviceNameNamespaceOwnerID(om *informer.ObjectMeta, ownerName s
 	// and labels
 	if envName, ok := s.serviceNameFromEnv(ownerKey); ok {
 		serviceName = envName
+	} else if s.serviceNameTemplate != nil {
```

Comment:
I think this option should have priority over the `serviceNameFromEnv`, applying the logic of "user defined > automatic"

Reply:
I actually thought about it, but left it as is, because the serviceNameFromEnv is more user defined than the template.
 
- annotation in pod
- env in deployment (owner)
- definition in beyla